src/frequenz/client/common/microgrid/components/components.py

```diff
@@ -0,0 +1,74 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""Defines the components that can be used in a microgrid."""
+from __future__ import annotations
+
+from enum import Enum
+
+# pylint: disable=no-name-in-module
+from frequenz.api.common.v1.microgrid.components.components_pb2 import (
+    ComponentCategory as PBComponentCategory,
+)
+
+# pylint: enable=no-name-in-module
+
+
+class ComponentCategory(Enum):
+    """Possible types of microgrid component."""
+
+    NONE = PBComponentCategory.COMPONENT_CATEGORY_UNSPECIFIED
+    """Unspecified component category."""
+
+    GRID = PBComponentCategory.COMPONENT_CATEGORY_GRID
+    """Grid component."""
+
+    METER = PBComponentCategory.COMPONENT_CATEGORY_METER
+    """Meter component."""
+
+    INVERTER = PBComponentCategory.COMPONENT_CATEGORY_INVERTER
+    """Inverter component."""
+
+    BATTERY = PBComponentCategory.COMPONENT_CATEGORY_BATTERY
+    """Battery component."""
+
+    EV_CHARGER = PBComponentCategory.COMPONENT_CATEGORY_EV_CHARGER
+    """EV charger component."""
+
+    CHP = PBComponentCategory.COMPONENT_CATEGORY_CHP
+    """CHP component."""
+
+
+def _component_category_from_protobuf(
```

Comment:
Since we are starting from scratch here, I would add a `@classmethod` `from_proto()` to the `enum` as we do with classes.

I'm not sure if I mentioned this, but I was thinking that if the idea is not to expose the protobuf stuff, when it might be better to have the to/from proto conversion separate (even in a separate module, so protobuf stuff is not imported at all when only using the wrappers), but then for enums that would be super inconvenient because we wouldn't be able to use the same values as in the protobuf stuff, which could be very useful when debugging (or we'll need to duplicate everything manually), so I guess for a first version it might make more sense to keep it simple (and coupled).